README.md
# ELS - Enterprise Linux/EuroLinux Sources - shotwell
 
## Usage:
  Checkout branch or tag. Use `make sources` to recreate  SOURCE0 archive.


src/ChangeLog
commit 921ff14cdaf58f804ab1f5a2f9abd15ee5ffa19c
Author: Jens Georg <mail@jensge.org>
Date:   Sat Jul 14 13:06:00 2018 +0200

    Prepare Shotwell 0.28.4

M	NEWS
M	configure.ac
M	meson.build

commit ce6d50975ffd7741f4b30a1b144e02eb4e868da9
Author: Jens Georg <mail@jensge.org>
Date:	Thu Jun 28 22:36:38 2018 +0200

    camera: Temporarily remove libusb work-around

    Work-around for !4 - unbreaking PTP support for most users is probably
    more important than catering this somewhat obscure
    (obsolete?) behavior

M	src/camera/CameraTable.vala

commit 9e12fe7a5af9eda05d8c5e6d0f771b1c8021d18d
Author: Jens Georg <mail@jensge.org>
Date:	Thu Jun 28 22:03:56 2018 +0200

    camera: Add mtp to supported mount schemes

    This allows automatic unmounting of MTP devices.

M	src/library/LibraryWindow.vala

commit 91f755dafd82d53aa0e2753fcfc8eedd8fce8918
Author: Piotr Drąg <piotrdrag@gmail.com>
Date:	Tue Jun 5 15:24:46 2018 +0200

    Fix keywords in Turkish translation


    (cherry picked from commit 64c40c6f0fe4558ec1bab4851cc771991593adda)

M	po/tr.po

commit 5bb67119597371a34b6de52e7633082919a981cf
Author: Jens Georg <mail@jensge.org>
Date:	Fri Jun 8 11:31:59 2018 +0200

    google: Fix OAuth2 token extraction

    The old code parsed the page title which previously was the URI. That
    seems to have changed lately to be empty.

    Use proper OAuth2 redirect_uri and handle the custom Scheme just
    like in
    Flickr for OAuth1.

    Closes #12

M	plugins/authenticator/shotwell/GoogleAuthenticator.vala

commit 4d55d976d0d458de0770616b934b3deace519c0f
Author: Jens Georg <mail@jensge.org>
Date:	Wed May 23 23:29:43 2018 +0200

    Fix slideshow settings dialog

    - Missing response codes for buttons
    - Wrong template class name

    https://bugzilla.gnome.org/show_bug.cgi?id=796370

M	ui/slideshow_settings.ui

commit cacc8bd5415d978f2eb64f215e4be696290c6edc
Author: Jens Georg <mail@jensge.org>
Date:	Mon May 7 19:37:12 2018 +0200

    Prepare 0.28.3

M	NEWS
M	configure.ac
M	meson.build

commit 50a532560e1627cae66d75036faab55e3247926b
Author: Piotr Drąg <piotrdrag@gmail.com>
Date:	Sun May 6 18:58:26 2018 +0200

    Update Polish translation

M	po/pl.po

commit 93d14085fc7deb433bc018e3a9927c1eb252a450
Author: Jens Georg <mail@jensge.org>
Date:	Sun May 6 15:11:31 2018 +0200

    Fix lined tags

    Fix regression introduced by a5ba963889c733d6ac80dad9a09058d0a214eb94

M	src/CheckerboardLayout.vala

commit 828981065e81d04b8160e131ca2c6c8ef91dd2fe
Author: Jens Georg <mail@jensge.org>
Date:	Sun Apr 29 11:17:19 2018 +0200

    checkerboard: Translate coordinates while drawing

    This prevents that we exceed the translation matrix limits in pixman,
    the root cause for the Out of memory errors seen on HiDPI displays

    https://bugzilla.gnome.org/show_bug.cgi?id=786702

M	src/CheckerboardLayout.vala

commit 776da8bf0c62d016273e1711e8874a2d7bef09d9
Author: Jens Georg <mail@jensge.org>
Date:	Sat Apr 14 18:19:32 2018 +0200

    Fix spacing for flagged button

M	src/SearchFilter.vala

commit 77b572462178a1f85633d315e0da69ab51b58f79
Author: Jens Georg <mail@jensge.org>
Date:	Fri Apr 13 19:29:29 2018 +0200

    Fix spacing in search toolbar

M	src/SearchFilter.vala

commit 4125c3461149ddec7fb58f58390a5c052eb7f1fc
Author: Jens Georg <mail@jensge.org>
Date:	Mon Apr 9 20:26:42 2018 +0200

    Prepare 0.28.2

M	NEWS
M	configure.ac
M	meson.build

commit 717552104d580c9dad3c8953085fa3e08b26c750
Author: Jens Georg <mail+gitkraken@jensge.org>
Date:	Sat Mar 31 13:50:00 2018 +0200

    Piwigo: Fix logging in with "&" in password

    https://bugzilla.gnome.org/show_bug.cgi?id=718742

M	plugins/shotwell-publishing/PiwigoPublishing.vala

commit 986cb24554aa9dfd54634b087ee86e7ec2f8c586
Author: Jens Georg <mail@jensge.org>
Date:	Wed Mar 28 19:19:42 2018 +0200

    Fix crash on editing images with alpha channel

    The cache does not carry alpha channel, use proper counting there
    to not
    exceed the cache size

M	src/ColorTransformation.vala

commit 6c1939ef77f33a08204a0869572be62db98635f4
Author: Jens Georg <mail@jensge.org>
Date:	Mon Mar 26 00:29:14 2018 +0200

    saved search: Fix working with Gtk 3.18

    Looks like Glade's verifier did not find this.

    https://bugzilla.gnome.org/show_bug.cgi?id=794673

M	ui/saved_search_dialog.ui

commit 46483b7a7997555d33336177b7f5b9a58cc94db1
Author: Jens Georg <mail@jensge.org>
Date:	Sun Mar 25 14:48:21 2018 +0200

    Prepare 0.28.1

M	NEWS
M	configure.ac
M	meson.build

commit 119d0f3b29c5c017081c014af14fcd609bbaa2d5
Author: Jens Georg <mail@jensge.org>
Date:	Sun Mar 25 14:34:46 2018 +0200

    Revert "publishers: Properly escape urlencoded payload"

    It Breaks at least flicker upload

    This reverts commit 7dc86b74c8a4cd251fb8815a818a206cd7a0d6b0.

M	plugins/common/RESTSupport.vala

commit caaacf4a1c673ca3ea778251c12577274e736859
Author: gogo <trebelnik2@gmail.com>
Date:	Thu Mar 22 00:05:12 2018 +0000

    Update Croatian translation

M	po/hr.po

commit c723f2c1a17568272623340ca4cf696897171948
Author: Jens Georg <mail@jensge.org>
Date:	Mon Mar 12 22:50:32 2018 +0100

    Fix undefined symbol error

    Or: inline does not do what you expect it to do.

    https://en.wikipedia.org/wiki/Inline_function#C99

    We really want static inline here.

    https://bugzilla.gnome.org/show_bug.cgi?id=794293

M	src/_transformation.c

commit 3abcc2f5061e27ba5b5b244ad5e059ba7c3eb2f2
Author: Rūdolfs Mazurs <rudolfsm@src.gnome.org>
Date:	Thu Mar 15 22:20:17 2018 +0200

    Update Latvian translation

M	po/lv.po

commit ecaf52c6dddb02a0c7c92c96ae9869025037465b
Author: Jens Georg <mail@jensge.org>
Date:	Mon Mar 12 22:06:46 2018 +0100

    Prepare 0.28.0

M	NEWS
M	configure.ac
M	icons/Makefile.am
D	icons/about-aachen.jpg
A	icons/about-braunschweig.jpg
M	meson.build
M	org.gnome.Shotwell.gresource.xml
M	src/AppWindow.vala
M	src/Resources.vala

commit 1c0d3d0b0fe6f5f244f126c75cae881e5b5cbc2a
Author: Alan Mortensen <alanmortensen.am@gmail.com>

src/src/util/ui.c
/* ui.c generated by valac 0.40.4, the Vala compiler
 * generated from ui.vala, do not modify */

/* Copyright 2016 Software Freedom Conservancy Inc.
 *
 * This software is licensed under the GNU Lesser General Public License
 * (version 2.1 or later).  See the COPYING file in this distribution.
 */


#include <glib.h>
#include <glib-object.h>
#include "shotwell-plugin-dev-1.0.h"
#include <stdlib.h>
#include <string.h>
#include <gtk/gtk.h>
#include <gdk/gdk.h>


#define TYPE_ADJUSTMENT_RELATION (adjustment_relation_get_type ())

#define TYPE_COMPASS_POINT (compass_point_get_type ())

#define TYPE_DIRECTION (direction_get_type ())

typedef enum  {
	ADJUSTMENT_RELATION_BELOW,
	ADJUSTMENT_RELATION_IN_RANGE,
	ADJUSTMENT_RELATION_ABOVE
} AdjustmentRelation;

typedef enum  {
	COMPASS_POINT_NORTH,
	COMPASS_POINT_SOUTH,
	COMPASS_POINT_EAST,
	COMPASS_POINT_WEST
} CompassPoint;

typedef enum  {
	DIRECTION_FORWARD,
	DIRECTION_BACKWARD
} Direction;



GType adjustment_relation_get_type (void) G_GNUC_CONST;
GType compass_point_get_type (void) G_GNUC_CONST;
GType direction_get_type (void) G_GNUC_CONST;
SpitTransitionsDirection direction_to_transition_direction (Direction self);
const gchar* direction_to_string (Direction self);
void spin_event_loop (void);
AdjustmentRelation get_adjustment_relation (GtkAdjustment* adjustment,
                                            gint value);
void get_adjustment_page (GtkAdjustment* hadj,
                          GtkAdjustment* vadj,
                          GdkRectangle* result);
gboolean has_only_key_modifier (GdkModifierType field,
                                GdkModifierType mask);


GType
adjustment_relation_get_type (void)
{
	static volatile gsize adjustment_relation_type_id__volatile = 0;
	if (g_once_init_enter (&adjustment_relation_type_id__volatile)) {
		static const GEnumValue values[] = {{ADJUSTMENT_RELATION_BELOW, "ADJUSTMENT_RELATION_BELOW", "below"}, {ADJUSTMENT_RELATION_IN_RANGE, "ADJUSTMENT_RELATION_IN_RANGE", "in-range"}, {ADJUSTMENT_RELATION_ABOVE, "ADJUSTMENT_RELATION_ABOVE", "above"}, {0, NULL, NULL}};
		GType adjustment_relation_type_id;
		adjustment_relation_type_id = g_enum_register_static ("AdjustmentRelation", values);
		g_once_init_leave (&adjustment_relation_type_id__volatile, adjustment_relation_type_id);
	}
	return adjustment_relation_type_id__volatile;
}


GType
compass_point_get_type (void)
{
	static volatile gsize compass_point_type_id__volatile = 0;
	if (g_once_init_enter (&compass_point_type_id__volatile)) {
		static const GEnumValue values[] = {{COMPASS_POINT_NORTH, "COMPASS_POINT_NORTH", "north"}, {COMPASS_POINT_SOUTH, "COMPASS_POINT_SOUTH", "south"}, {COMPASS_POINT_EAST, "COMPASS_POINT_EAST", "east"}, {COMPASS_POINT_WEST, "COMPASS_POINT_WEST", "west"}, {0, NULL, NULL}};
		GType compass_point_type_id;
		compass_point_type_id = g_enum_register_static ("CompassPoint", values);
		g_once_init_leave (&compass_point_type_id__volatile, compass_point_type_id);
	}
	return compass_point_type_id__volatile;
}


SpitTransitionsDirection
direction_to_transition_direction (Direction self)
{
	SpitTransitionsDirection result = 0;
#line 25 "/home/jens/Source/shotwell/src/util/ui.vala"
	switch (self) {
#line 25 "/home/jens/Source/shotwell/src/util/ui.vala"
		case DIRECTION_FORWARD:
#line 98 "ui.c"
		{
#line 27 "/home/jens/Source/shotwell/src/util/ui.vala"
			result = SPIT_TRANSITIONS_DIRECTION_FORWARD;
#line 27 "/home/jens/Source/shotwell/src/util/ui.vala"
			return result;
#line 104 "ui.c"
		}
#line 25 "/home/jens/Source/shotwell/src/util/ui.vala"
		case DIRECTION_BACKWARD:
#line 108 "ui.c"
		{
#line 30 "/home/jens/Source/shotwell/src/util/ui.vala"
			result = SPIT_TRANSITIONS_DIRECTION_BACKWARD;
#line 30 "/home/jens/Source/shotwell/src/util/ui.vala"
			return result;
#line 114 "ui.c"
		}
		default:
		{
			GEnumValue* _tmp0_;
#line 33 "/home/jens/Source/shotwell/src/util/ui.vala"
			_tmp0_ = g_enum_get_value (g_type_class_ref (TYPE_DIRECTION), self);
#line 33 "/home/jens/Source/shotwell/src/util/ui.vala"
			g_error ("ui.vala:33: Unknown Direction %s", (_tmp0_ != NULL) ? _tmp0_->value_name : NULL);
#line 123 "ui.c"
		}
	}
}


GType
direction_get_type (void)
{
	static volatile gsize direction_type_id__volatile = 0;
	if (g_once_init_enter (&direction_type_id__volatile)) {
		static const GEnumValue values[] = {{DIRECTION_FORWARD, "DIRECTION_FORWARD", "forward"}, {DIRECTION_BACKWARD, "DIRECTION_BACKWARD", "backward"}, {0, NULL, NULL}};
		GType direction_type_id;
		direction_type_id = g_enum_register_static ("Direction", values);
		g_once_init_leave (&direction_type_id__volatile, direction_type_id);
	}
	return direction_type_id__volatile;
}


void
spin_event_loop (void)
{
#line 39 "/home/jens/Source/shotwell/src/util/ui.vala"
	while (TRUE) {
#line 39 "/home/jens/Source/shotwell/src/util/ui.vala"
		if (!gtk_events_pending ()) {
#line 39 "/home/jens/Source/shotwell/src/util/ui.vala"
			break;
#line 152 "ui.c"
		}
#line 40 "/home/jens/Source/shotwell/src/util/ui.vala"
		gtk_main_iteration ();
#line 156 "ui.c"
	}
}


AdjustmentRelation
get_adjustment_relation (GtkAdjustment* adjustment,
                         gint value)
{
	AdjustmentRelation result = 0;
#line 43 "/home/jens/Source/shotwell/src/util/ui.vala"
	g_return_val_if_fail (GTK_IS_ADJUSTMENT (adjustment), 0);
#line 44 "/home/jens/Source/shotwell/src/util/ui.vala"
	if (value < ((gint) gtk_adjustment_get_value (adjustment))) {
#line 45 "/home/jens/Source/shotwell/src/util/ui.vala"
		result = ADJUSTMENT_RELATION_BELOW;
#line 45 "/home/jens/Source/shotwell/src/util/ui.vala"
		return result;
#line 174 "ui.c"
	} else {
#line 46 "/home/jens/Source/shotwell/src/util/ui.vala"
		if (value > ((gint) (gtk_adjustment_get_value (adjustment) + gtk_adjustment_get_page_size (adjustment)))) {
#line 47 "/home/jens/Source/shotwell/src/util/ui.vala"
			result = ADJUSTMENT_RELATION_ABOVE;
#line 47 "/home/jens/Source/shotwell/src/util/ui.vala"
			return result;
#line 182 "ui.c"
		} else {
#line 49 "/home/jens/Source/shotwell/src/util/ui.vala"
			result = ADJUSTMENT_RELATION_IN_RANGE;
#line 49 "/home/jens/Source/shotwell/src/util/ui.vala"
			return result;
#line 188 "ui.c"
		}
	}
}


void
get_adjustment_page (GtkAdjustment* hadj,
                     GtkAdjustment* vadj,
                     GdkRectangle* result)
{
	GdkRectangle rect = {0};
#line 52 "/home/jens/Source/shotwell/src/util/ui.vala"
	g_return_if_fail (GTK_IS_ADJUSTMENT (hadj));
#line 52 "/home/jens/Source/shotwell/src/util/ui.vala"
	g_return_if_fail (GTK_IS_ADJUSTMENT (vadj));
#line 53 "/home/jens/Source/shotwell/src/util/ui.vala"
	memset (&rect, 0, sizeof (GdkRectangle));
#line 54 "/home/jens/Source/shotwell/src/util/ui.vala"
	rect.x = (gint) gtk_adjustment_get_value (hadj);
#line 55 "/home/jens/Source/shotwell/src/util/ui.vala"
	rect.y = (gint) gtk_adjustment_get_value (vadj);
#line 56 "/home/jens/Source/shotwell/src/util/ui.vala"
	rect.width = (gint) gtk_adjustment_get_page_size (hadj);
#line 57 "/home/jens/Source/shotwell/src/util/ui.vala"
	rect.height = (gint) gtk_adjustment_get_page_size (vadj);
#line 59 "/home/jens/Source/shotwell/src/util/ui.vala"
	*result = rect;
#line 59 "/home/jens/Source/shotwell/src/util/ui.vala"
	return;
#line 218 "ui.c"
}


gboolean
has_only_key_modifier (GdkModifierType field,
                       GdkModifierType mask)
{
	gboolean result = FALSE;
#line 77 "/home/jens/Source/shotwell/src/util/ui.vala"
	result = (field & ((((((((GDK_SHIFT_MASK | GDK_CONTROL_MASK) | GDK_MOD1_MASK) | GDK_MOD3_MASK) | GDK_MOD4_MASK) | GDK_MOD5_MASK) | GDK_SUPER_MASK) | GDK_HYPER_MASK) | GDK_META_MASK)) == mask;
#line 77 "/home/jens/Source/shotwell/src/util/ui.vala"
	return result;
#line 231 "ui.c"
}



